testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -683,6 +714,19 @@ impl BlockMinerThread {
             });
         };
 
+        if self.mined_blocks.is_empty() {
+            // We could call this even if self.mined_blocks was not empty, but would return the same value, so save the effort and only do it when necessary.
+            // If we are starting a new tenure, then make sure we are building off of the last block of our parent tenure
+            if let Some(last_tenure_finish_block_header) =
+                NakamotoChainState::get_nakamoto_tenure_finish_block_header(
+                    chain_state.db(),
+                    &stacks_tip_header.consensus_hash,
+                )
+                .expect("FATAL: could not query parent tenure finish block")
+            {
+                stacks_tip_header = last_tenure_finish_block_header;
+            }
+        }
```

Comment:
I think it makes it more clear if the logic of like 689-729 is combined so that first we get the tenure consensus hash of the parent we're building on, and then we always fetch the finish block header.

```rust
let tenure_of_parent = if let Some(block) = self.mined_blocks.last() {
    block.header.consensus_hash.clone()
} else {
    let Some(parent_tenure_block) = NakamotoChainState::get_block_header(chain_state.db(), &self.parent_tenure_id) else {
        debug!("No Stacks chain tip known, will return a genesis block");
        ...
    };
    parent_tenure_block.consensus_hash
};

let Some(parent_block_header) = NakamotoChainState::get_nakamoto_tenure_finish_block_header(..).expect("FATAL: ...") else {
    warn!("Failed to load the last block of the parent tenure");
    return Err(NakamotoNodeError::...)
};
```

Note: I also suggest renaming `stacks_tip_header` to `parent_block_header`. The parent block is no longer _necessarily_ the stacks tip (even though we usually expect it to be so).

Reply:
I can prob rework it, but pretty sure one of the reasons I did it this way is that it could be the first nakamoto tenure ever and therefore its parent tenure tip check will fail as it has no nakamoto parent. This enforces that it uses the genesis block in that case. Otherwise I think we will error on the second block in the first nakamoto tenure? Will give it a try though.

Reply:
K I can't get this to work. It seems that there is no nakamoto tenure stored for the case of the first nakamoto tenure. i.e. the tenure is not empty, It is not the first block. it has a valid prior block so it doens't use a genesis block...BUT it is part of the first tenure still. therefore its not stored in the last completed list of nakamoto tenures it seems and it therefore hits "Failed to load the last block of the parent tenure".